File: hmwk_02.java
//Name:Amey Puranik
//Netid - axp1390
//Date : 03/27/2018

import java.nio.file.Paths;
import java.nio.file.Files;
import java.util.Arrays;

public class hmwk_02{
public static void processToken( String token ) {
   String inttoken= "^[0][dD][0-9]+$";
   String Flotingpoint= "^[0-9]+\\.[0-9]+$";
   String ID= "^[_a-mN-Z][n-zA-M_0-9]*$";
   String EFP= "^[0-9][0-9]*[eE+-]+[0-9]+$||^[0-9]*\\.[0-9]*[eE+-]*[0-9]*$";
   String HX= "^[0][xX][a-fA-F0-9]+$";
   
   if (token.matches(inttoken)) {
	   
	   System.out.println( ">" + token + "< Matches INT." );
	   
                                 }
   
   else if (token.matches(Flotingpoint)) {
	   
	   System.out.println( ">" + token + "< Matches FP." );
	   
                                 }
   
   else if (token.matches(ID)) {
	   
	   System.out.println( ">" + token + "< Matches ID." );
	   
                                 }
   
   else if (token.matches(HX)) {
	   
	   System.out.println( ">" + token + "< Matches HEXINT." );
	   
                                 }
   else if (token.matches(EFP)) {
	   
	   System.out.println( ">" + token + "< Matches EFP." );
	   
                                 }
   else {

          System.out.println( ">" + token + "< does not match." );
      }	

 
}

public static void main( String[] args ) {
 System.out.println( "processing tokens from " + args[ 0 ] + " ..." );

 try {
   Files.lines( Paths.get( args[ 0 ] ) ).forEachOrdered(
     line -> Arrays.stream( line.split( "\\s+" )  )
         .forEachOrdered( token -> processToken( token ) ) );
 } catch ( java.io.IOException e ) {
   e.printStackTrace();
 }
}
}

File: hmwk_02.py
# Amey Jayant Puranik
# axp1390
# 03-28-2018
#--------------------------------------------------
import re
import sys

#--------------------------------------------------
def processToken( token ) :
  global integ
  global fp
  global ID
  global Hexnt
  global EFP
  integ ='false'
  pattern=re.compile("^[0][dD][0-9]+$")
  integ=pattern.match(token)
  fp='false'
  pattern=re.compile("^[0-9]+\\.[0-9]+$")
  fp=pattern.match(token)
  ID='false'
  pattern=re.compile("^[_a-mN-Z][n-zA-M_0-9]*$")
  ID=pattern.match(token)
  Hexnt='false'
  pattern=re.compile("^[0][xX][a-fA-F0-9]+$")
  Hexnt=pattern.match(token)
  EFP='false'
  pattern=re.compile("^[0-9][0-9]*[eE+-]+[0-9]+$|^[0-9]*\\.[0-9]*[eE+-]*[0-9]*$")
  EFP=pattern.match(token)
  if integ:
    print( '>%s< Matches INT' % ( token ) )
  elif fp: 
    print( '>%s< Matches FP' % ( token ) )
  elif ID: 
    print( '>%s< Matches ID ' % ( token ) )
  elif Hexnt: 
    print( '>%s< Matches HEXINT ' % ( token ) )
  elif EFP: 
    print( '>%s< Matches EFP ' % ( token ) )
  else :
    print( '>%s< Does not match ' % ( token ) )
#--------------------------------------------------
def main() :
  fName = sys.argv[ 1 ]
  print( 'processing tokens from ' + fName + ' ...' )

  with open( fName, 'r' ) as fp :
    lines = fp.read().replace( '\r', '' ).split( '\n' )

  for line in lines :
    for token in line.split() :
      processToken( token )

#--------------------------------------------------
if ( __name__ == '__main__' ) :
  main()

#--------------------------------------------------
